File: HireHubDemo/hirehub-core/src/main/java/com/hirehub/core/dtos/ApplicantDTO.java
package com.hirehub.core.dtos;

import com.hirehub.core.entities.JobApplication;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class ApplicantDTO {

    private Long applicationId; // application id
    private Long candidateId; // ✅ REQUIRED to open profile
    private String candidateName; // candidate name
    private String email;
    private String skills;
    private JobApplication.Status status;

    // Referral tracking
    private Long referredByEmployeeId;
    private String referredByEmployeeName;
}

File: HireHubDemo/hirehub-referral/Dtos/ApplicantDto.cs
namespace HireHub.Referral.Dtos
{
    public class ApplicantDto
    {
        public long Id { get; set; }
        public long CandidateId { get; set; }
        public string CandidateName { get; set; }
        public string Email { get; set; }
        public string Skills { get; set; }
        public string Status { get; set; }
        
        // Referral tracking
        public long? ReferredByEmployeeId { get; set; }
        public string ReferredByEmployeeName { get; set; }
    }
}
